src/imitation/algorithms/bc.py

```diff
@@ -208,75 +205,30 @@ def train(
                 batch_dict = self.expert_dataset.sample(self.batch_size)
                 assert len(batch_dict["obs"]) == self.batch_size
                 samples_so_far += self.batch_size
-                feed_dict = {
-                    self._true_acts_ph: batch_dict["acts"],
-                    self.policy.obs_ph: batch_dict["obs"],
-                }
-                _, stats_dict = self.sess.run(
-                    [self._train_op, self._stats_dict], feed_dict=feed_dict
-                )
+
+                obs_tensor = th.as_tensor(batch_dict["obs"]).to(self.policy.device)
```

Comment:
@qxcv Does querying `self.policy.device` as I've done in this line of code seem like the best decision to you? I'm wondering if this will break if Stable Baselines3 removes device from its BasePolicy constructor as you suggested in your bug report.